.github/workflows/visual-regression.yml

```
@@ -0,0 +1,59 @@
+name: Visual Regressions
+
+on:
+  workflow_call:
+    inputs:
+      repository:
+        required: true
+        type: string
+
+env:
+  VISUAL_REGRESSION_SNAPSHOT_DIRECTORY: '__image_snapshots__'
+
+jobs:
+  test:
+    name: Run Tests
+    runs-on: ubuntu-latest
+    if: github.event.ref != 'refs/heads/main'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'npm'
+      - name: 'Download artifacts'
+        # Turns out you can not use artifacts from a previous workflow run.
+        # I'm reluctant to check in the resulting screenshots into the repository,
+        # which is why we use the following hack, until it's solved on Github side:
+        # https://github.com/actions/download-artifact/issues/3
+        run: |
+          RUN_ID=`gh run --repo ${{ inputs.repository }} --branch main list --workflow "Visual Regressions" --json databaseId --jq .[0].databaseId`
+          echo "Downloading snapshots created in run ${RUN_ID}"
+          gh run --repo ${{ inputs.repository }} download ${RUN_ID} -n "visual-regression-snapshots" -D ${{ env.VISUAL_REGRESSION_SNAPSHOT_DIRECTORY }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

I'd recommend to redirect the input parameter `repository` through an environment variable, just to make sure that it can't be used for injecting shell commands. See [GitHub's docs section about the mitigation _"Using an intermediate environment variable"_](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable)

Additionally, I think you should be able to access `VISUAL_REGRESSION_SNAPSHOT_DIRECTORY` as an environment variable rather than a context value (so that it's evaluated by the shell and not by the CI runner):

```suggestion
          RUN_ID=`gh run --repo $REPOSITORY --branch main list --workflow "Visual Regressions" --json databaseId --jq .[0].databaseId`
          echo "Downloading snapshots created in run ${RUN_ID}"
          gh run --repo $REPOSITORY download ${RUN_ID} -n "visual-regression-snapshots" -D $VISUAL_REGRESSION_SNAPSHOT_DIRECTORY
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          REPOSITORY: ${{ inputs.repository }}
```